### ISSUE TITLE
Calls to https://api.botframework.com/api/usertoken/GetToken returning 404

### ISSUE THREAD

**Issue by davidgerrard** [microsoft/botbuilder-dotnet#6567]
## Version
Failing on 4.18.1 and 4.16.0

## Describe the bug
As part of our bot implementation we use the OAuthPrompt to obtain a token from our identity server, so the bot can call our APIs.

We haven't made any changes to the bot recently, but have discovered over the last 48 hours, a number of users are constantly being asked to Log In via the prompt card. When the next API request is made and we check with the OAuthPrompt for a token, it makes a request to:

 https://api.botframework.com/api/usertoken/GetToken with the following query params:
userId= the MS Teams User Id,
connectionName= our OAuth connection name,
channelId=msteams

All these values look valid, but the GET request returns a 404. This then leads to a GET request being made to:

https://api.botframework.com/api/botsignin/GetSignInResource, with a query param 'state' which is the Teams User JWT. This comes back with a 200 OK.

This then prompts the user to click Log In again to get another token. Our identity server provides another token and we can make a successful API call. The next API call we make (seconds later) we are prompted again to login as we get another 404 from the GetToken endpoint.

Looking in the code it seems to be in the class 'UserTokenAccess', that the 'GetUserTokenAsync' method is returning null. If it is called again with a magic code in the query params, it succeeds.

`return await adapter.GetUserTokenAsync(turnContext, settings.OAuthAppCredentials, settings.ConnectionName, magicCode, cancellationToken).ConfigureAwait(false);`

## To Reproduce
1. Use the OAuthPrompt to prompt to login
2. Log in to your configured provider
3. Check and use the JWT in the TokenResponse
4. Call the OAuthPrompt again to get the JWT
5. Observe that you don't get the current token, but are asked to log in again

## Expected behavior
Once you have logged in via the OAuthPrompt, you can call the prompt again to retrieve the same valid token

**Comment by davidgerrard** [comment]
As an additional note, you can reproduce the issue with the sample here: https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/18.bot-authentication

Follow the sample to get it up and running.
Sign in
Click Yes when asked to view your token
You are prompted to sign in again.

**Comment by ramfattah** [comment]
Hi @davidgerrard, we are investigating.

**Comment by ramfattah** [comment]
Hi @davidgerrard, 

I was not able to reproduce this issue whent tested with sample [18.bot-authentication](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/18.bot-authentication) with [AAD 2 Identity provider](https://learn.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=userassigned%2Caadv2%2Ccsharp#:~:text=See%20next%20section.-,Register%20the%20Azure%20AD%20identity%20provider%20with%20the%20bot,-The%20next%20step).

After login from initial OAuthPrompt, I receive the same token.


https://user-images.githubusercontent.com/38049078/207181506-9e59a53e-cde6-4e96-969e-8a9333a2eef1.mp4

#

**Comment by ramfattah** [comment]
Hi @davidgerrard, 

I was not able to reproduce this issue when tested with sample [18.bot-authentication](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/18.bot-authentication) with [AAD 2 Identity provider](https://learn.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=userassigned%2Caadv2%2Ccsharp#:~:text=See%20next%20section.-,Register%20the%20Azure%20AD%20identity%20provider%20with%20the%20bot,-The%20next%20step).

After login from initial OAuthPrompt, I receive the same token.


https://user-images.githubusercontent.com/38049078/207181506-9e59a53e-cde6-4e96-969e-8a9333a2eef1.mp4

**Comment by davidgerrard** [comment]
Yes, it looks as though it has 'fixed itself' over the last 24 hours. I can only assume the issue in the bot framework token service has been resolved. Would be nice to get a response from that MS team, but happy to close this.
